### ISSUE TITLE
Text input with "display: inline" has takes up no horizontal space

### ISSUE THREAD

**Issue by jdm** [servo/servo#33669]
```html
<input style="display: inline"></input>
```
Firefox:
<img width="387" alt="Screenshot 2024-10-05 at 9 20 06 PM" src="https://github.com/user-attachments/assets/9a08fce1-f5cf-41d4-aa1a-25c139bcf1b2">
Servo:
<img width="212" alt="Screenshot 2024-10-05 at 9 20 17 PM" src="https://github.com/user-attachments/assets/7df1c265-8295-43c8-a11e-45adeca56abe">

**Comment by jdm** [comment]
The default display value for input fields is inline-block: https://github.com/servo/servo/blob/main/resources/user-agent.css#L360

**Comment by jdm** [comment]
This affects `<textarea>` the same way.

**Comment by Loirooriol** [comment]
https://html.spec.whatwg.org/multipage/rendering.html#form-controls

> For [input](https://html.spec.whatwg.org/multipage/input.html#the-input-element) elements where the [type](https://html.spec.whatwg.org/multipage/input.html#attr-input-type) attribute is not in the [Hidden](https://html.spec.whatwg.org/multipage/input.html#hidden-state-(type=hidden)) state or the [Image Button](https://html.spec.whatwg.org/multipage/input.html#image-button-state-(type=image)) state, and that are [being rendered](https://html.spec.whatwg.org/multipage/rendering.html#being-rendered), are expected to act as follows:
> 
>   -  The [inner display type](https://drafts.csswg.org/css-display/#inner-display-type) is always 'flow-root'.

So it should stay as an inline-block even with `display: inline`.

**Comment by PotatoCP** [comment]
I'd like to try this

**Comment by PotatoCP** [comment]
@Loirooriol do you think we should also apply the same behavior for textarea (display inline always 'flow-root')?

**Comment by Loirooriol** [comment]
Yes: https://html.spec.whatwg.org/multipage/rendering.html#the-textarea-element-2


> The [textarea](https://html.spec.whatwg.org/multipage/form-elements.html#the-textarea-element) element is a [devolvable widget](https://drafts.csswg.org/css-ui/#devolvable) expected to render as an ['inline-block'](https://drafts.csswg.org/css2/#value-def-inline-block) box depicting a multiline text control.
